libres/lib/analysis/ies/ies_data.cpp

```diff
@@ -48,38 +42,28 @@ int ies::data::Data::iteration_nr() const { return this->m_iteration_nr; }
 
 int ies::data::Data::inc_iteration_nr() { return ++this->m_iteration_nr; }
 
-void ies::data::Data::update_ens_mask(const bool_vector_type *mask) {
-    if (this->m_ens_mask)
-        bool_vector_free(this->m_ens_mask);
-
-    this->m_ens_mask = bool_vector_alloc_copy(mask);
+void ies::data::Data::update_ens_mask(const std::vector<bool> &mask) {
+    this->m_ens_mask = mask;
 }
 
 int ies::data::Data::ens_size() const { return this->m_ens_size; }
 
-void ies::data::Data::store_initial_obs_mask(const bool_vector_type *mask) {
-    if (!this->m_obs_mask0)
-        this->m_obs_mask0 = bool_vector_alloc_copy(mask);
+void ies::data::Data::store_initial_obs_mask(const std::vector<bool> &mask) {
+    if (this->m_obs_mask0.size() == 0)
```

Comment:
Instead of comparing `size() == 0` the idiomatic way to is to check `empty()`:
```C++
if (this->m_obs_mask0.empty())
    this->m_obs_mask0 = mask;
```